sympy/simplify/tests/test_cse.py

```diff
@@ -477,3 +477,24 @@ def check(eq):
 def test_hollow_rejection():
     eq = [x + 3, x + 4]
     assert cse(eq) == ([], eq)
+
+
+def test_cse_ignore():
+    exprs = [exp(y)*(3*y + 3*sqrt(x+1)), exp(y)*(5*y + 5*sqrt(x+1))]
+    subst1, red1 = cse(exprs)
+    for symb, sub in subst1:
+        if y in sub.free_symbols:
+            break  # expected behaviour
+    else:
+        raise ValueError("cse failed to identify any term with y")
```

Comment:
It would be better to use `AssertionError` here. In the loop below you can just write `assert y not in sub.free_symbols, "Sub-expressions containing y must be ignored"`. You could also rewrite this as 

```py
assert any(y in sub.free_symbols for symb, sub in subst1), "cse failed to identify any term with y"
```